lib/index.js

```diff
@@ -328,7 +334,14 @@ Segment.prototype.enqueue = function(path, msg, fn) {
   var url = 'https://' + this.options.apiHost + path;
   var headers = { 'Content-Type': 'text/plain' };
   msg = this.normalize(msg);
-  this.debug('enqueueing');
+
+  // Print a log statement when messages exceed the maximum size. In the future,
+  // we may consider dropping this event on the client entirely.
+  if (json.stringify(msg).length > MAX_SIZE) {
```

Comment:
Interesting. What's the best way to do this check then? We used the `Buffer` class for Node.js (https://github.com/segmentio/loosely-validate-event/pull/4/files#diff-168726dbe96b3ce427e7fedce31bb0bcR112), but we don't have this API available on the browser (at least not without polyfilling I think).

Reply:
Reading a bit more about this, I think assuming UTF-16 is ok here. So I can just change the number to to match it based on UTF-16. Since this is only warning for now, I think it's ok if it's not 100% exact. What do you think?

Reply:
I would just point out that messages are likely, in practice, going to be mostly ASCII. And ASCII takes less space in UTF-8 than it does in UTF-16?

I don't think perfection is required here. Any check would be better than none, and doing encoding conversions just seems a bit wasteful for a rough order-of-magnitude check?

Reply:
@ucarion do you have a reference to when it can be ASCII? reading up https://stackoverflow.com/questions/27253150/json-stringify-to-utf-8 and https://mathiasbynens.be/notes/javascript-encoding it seems like the strings should only be encoded in UTF-16 or UCS-2.

Agreed that an exact check is not required here, and doing conversions just for the the check would be wasteful.

Reply:
Ah! Sorry, by "ASCII" in that sentence I meant the first 128 characters of Unicode -- the only characters supported by the ASCII encoding.

UTF-8 can represent these characters using one byte, where UTF-16 uses 2 bytes for these characters. So I suspect that _in practice_, most of the messages people will send us will in fact take less space over the wire than they do in JavaScript memory.

Reply:
I ended up checking and `XMLHTTPRequest` encodes the data back to UTF-8 before sending it to the server : 

- [`send()` method](https://xhr.spec.whatwg.org/#the-send()-method)
  > set request body and extractedContentType to the result of [extracting](https://fetch.spec.whatwg.org/#concept-bodyinit-extract) body
  > > ##### USVString
  > > Set action to an action that runs UTF-8 encode on object

You can safely ignore this!

Reply:
Cool - so we'll proceed with the change as is!